iothub/device/src/ClientFactory.cs

```diff
@@ -201,7 +204,7 @@ internal static InternalClient Create(string hostname, IAuthenticationMethod aut
             var connectionStringBuilder = IotHubConnectionStringBuilder.Create(hostname, gatewayHostname, authenticationMethod);
 
             // Make sure client options is initialized with the correct transport setting.
-            EnsureOptionsIsSetup(connectionStringBuilder.Certificate, ref options);
+            EnsureOptionsIsSetup(connectionStringBuilder.Certificate, transportSettings.FirstOrDefault()?.Proxy, ref options);
```

Comment:
The `FirstOrDefault()` approach might not work for proxy as well as it does for the client certificates; eg. if my transport options are [mqtt_tcp, mqtt_ws], then firstOrDefault would look at the tcp settings for the proxy whereas the actual proxy setting would only be under the ws settings.
This could be a bit complicated to set up since I believe we set up the file upload transport handler before we've finalized on a transport protocol to use.
We set up file upload transport handler as a part of the client set up whereas the transport protocol will be chosen only once an attempt is made to open up the connection.

Reply:
How about this?

```csharp
transportSettings.Where(x => x.Proxy != null).FirstOrDefault()?.Proxy
```

Reply:
Should I also add an exclusion for `DefaultWebProxySettings.Instance`?